tools/test/test_cmake.py

```diff
@@ -0,0 +1,72 @@
+import contextlib
+import os
+import typing
+from typing import Iterator, List, Optional
+import unittest
+import unittest.mock
+
+import tools.setup_helpers.env  # noqa: F401 unused but resolves circular import
+import tools.setup_helpers.cmake
+
+
+class TestCMake(unittest.TestCase):
+
+    def test_build_max_jobs(self) -> None:
+        with env_var('MAX_JOBS', '8'):
+            build_args = self._cmake_build_and_get_args()
+
+        self.assertListEqual(build_args[-3:], ['--', '-j', '8'])
+
+    def test_build_with_ninja(self) -> None:
+        with unittest.mock.patch.object(tools.setup_helpers.cmake, 'USE_NINJA', True):
+            build_args = self._cmake_build_and_get_args()
+
+        self.assertNotIn('--', build_args)
```

Comment:
We know exactly what the build args should be right? Seems like this could be a direct equality comparison to that instead of checking for individual elements

Reply:
well i only want to test one aspect of the args: namely whether or not the jobs are present. Encoding all the build args will make the test more brittle in the event of other changes.

I should probably *only* verify that `'-j'` is not present. What do you think?

Reply:
That makes sense to me